Add model evaluation documentation with W&B Weave and Tables

johndmulhausen opened the pull request:
## Summary

This PR adds comprehensive documentation for evaluating models using W&B's evaluation tools, focusing on both W&B Weave and W&B Tables.

Resolves DOCS-1024

## Changes

### New Documentation
- **`content/en/guides/models/evaluate-models.md`**: A new guide that covers:
  - How to evaluate models using W&B Weave
  - Integration with W&B Models Registry
  - Custom scorers and judges
  - Batch evaluations
  - Using W&B Tables for evaluation analysis
  - Interactive error analysis and model comparison

### New Assets
- **`assets/images/weave/evals.png`**: Dashboard visualization for Weave evaluations

## Key Features Documented

1. **W&B Weave Integration**
   - Loading models from W&B Registry and evaluating them with Weave
   - Creating evaluation datasets and custom scorers
   - Logging evaluation results back to W&B Models
   - Advanced features like LLM judges and batch evaluations

2. **W&B Tables for Evaluation**
   - Comparing model predictions side-by-side
   - Tracking prediction changes across epochs
   - Interactive filtering and error analysis
   - Rich media support for evaluation data

## Why These Changes?

Model evaluation is a critical part of the ML workflow, and W&B provides powerful tools for this purpose. This documentation:
- Helps users understand how to use W&B's evaluation capabilities
- Provides practical examples and code snippets
- Shows the integration between W&B Models Registry and Weave
- Demonstrates best practices for model evaluation workflows

## Testing

The documentation includes:
- Complete code examples that users can run
- Links to external resources and tutorials
- Clear explanations of concepts and features

Please review the content for accuracy and clarity. Let me know if any adjustments are needed!

<!-- preview-links-comment -->
📄 **[View preview links for changed pages](https://github.com/wandb/docs/pull/1558#issuecomment-3215281687)**

johndmulhausen committed:
Add documentation for evaluating models with W&B Weave and Tables

- Add comprehensive guide for model evaluation using W&B Weave
- Include examples for loading models from Registry and evaluating them
- Document custom scorers, judges, and batch evaluations
- Add section on using W&B Tables for evaluation analysis
- Include image asset for Weave evaluation dashboard

johndmulhausen committed:
Update evaluation documentation for W&B Weave

- Refine description of W&B Weave's integration with W&B Models for model evaluation.
- Remove redundant phrasing to enhance clarity and readability.

mdlinville commented on content/en/guides/models/evaluate-models.md:
```diff
@@ -0,0 +1,241 @@
+* **Evaluation Datasets**: Structured test sets with ground truth for systematic evaluation
+* **Model Versioning**: Track and compare different versions of your models
+* **Detailed Tracing**: Debug model behavior with complete input/output traces
+* **Cost Tracking**: Monitor API costs and token usage across evaluations
```
Nit: Sentence case for the bold items :)

johndmulhausen replied:
Fixed, good call!

mdlinville commented on content/en/guides/models/evaluate-models.md:
```diff
@@ -0,0 +1,241 @@
+* **Detailed Tracing**: Debug model behavior with complete input/output traces
+* **Cost Tracking**: Monitor API costs and token usage across evaluations
+
+### Getting started: Evaluate a model from W&B Models Registry
```
```suggestion
### Getting started: Evaluate a model from W&B Registry
```
There are a few instances of this phrase

mdlinville commented on content/en/guides/models/evaluate-models.md:
```diff
@@ -0,0 +1,241 @@
+
+### Getting started: Evaluate a model from W&B Models Registry
+
+This example shows how to download a model from W&B Models Registry and evaluate it using Weave:
```
```suggestion
Download a model from W&B Models Registry and evaluate it using Weave:
```

mdlinville commented on content/en/guides/models/evaluate-models.md:
````diff
@@ -0,0 +1,241 @@
+results = await evaluation.evaluate(model)
+```
+
+### Integrating Weave evaluations with W&B Models
````
```suggestion
### Integrate Weave evaluations with W&B Models
```

mdlinville commented on content/en/guides/models/evaluate-models.md:
```diff
@@ -0,0 +1,241 @@
+
+### Integrating Weave evaluations with W&B Models
+
+The [Models and Weave Integration Demo](https://weave-docs.wandb.ai/reference/gen_notebooks/Models_and_Weave_Integration_Demo) shows the complete workflow for:
```
```suggestion
The [Models and Weave integration demo notebook](https://weave-docs.wandb.ai/reference/gen_notebooks/Models_and_Weave_Integration_Demo) shows the complete workflow:
```
And change the bold below from gerunds to imperative verbs maybe?

johndmulhausen replied:
Fixed, good call :)

mdlinville commented on content/en/guides/models/evaluate-models.md:
```diff
@@ -0,0 +1,241 @@
+3. **Logging results back to W&B**: Connect evaluation metrics to your model runs
+4. **Versioning evaluated models**: Save improved models back to the Registry
+
+Example of logging evaluation results to both Weave and W&B Models:
```
```suggestion
Log evaluation results to both Weave and W&B Models:
```

mdlinville commented on content/en/guides/models/evaluate-models.md:
````diff
@@ -0,0 +1,241 @@
+run.log({"evaluation_results": eval_table})
+```
+
+### Advanced evaluation workflows
````
I'm not sure it is ideal to have two separate Advanced headings. WDYT?

johndmulhausen replied:
There is only one in each major area (Weave vs. Tables) and I think the pattern of "here's the general idea, here's the hard mode" for each of the paths makes sense? Gave them better names to highlight the approach-specific nature of those sections

mdlinville reviewed (approved):
A few suggestions but I will go ahead and improve so you are not blocked :)

johndmulhausen committed:
Update content/en/guides/models/evaluate-models.md

Co-authored-by: Matt Linville <matt.linville@wandb.com>

johndmulhausen committed:
Update content/en/guides/models/evaluate-models.md

Co-authored-by: Matt Linville <matt.linville@wandb.com>

johndmulhausen committed:
Update content/en/guides/models/evaluate-models.md

Co-authored-by: Matt Linville <matt.linville@wandb.com>

johndmulhausen committed:
Update content/en/guides/models/evaluate-models.md

Co-authored-by: Matt Linville <matt.linville@wandb.com>

johndmulhausen committed:
Merge branch 'main' into add-evals-section

github-actions[bot] committed:
Optimised images with calibre/image-actions

johndmulhausen committed:
Update model evaluation documentation for consistency and clarity

- Standardized capitalization in key features and workflows sections.
- Revised section titles for improved clarity, changing "Advanced evaluation features" to "Advanced Weave features" and "Advanced evaluation workflows" to "Advanced table workflows".

johndmulhausen committed:
Merge branch 'main' into add-evals-section
